[charts] Add legend (#9024)
mui/mui-x: docs/data/charts/legend/legend.md

---
product: charts
title: Charts - Legend
---

# Charts - Legend

<p class="description">Legend is the UI element mapping symbols and colors to the series' label.</p>

## Basic display

In chart components, the legend links series with `label` properties and their color.

{{"demo": "BasicLegend.js", "bg": "inline"}}

## Placement

The legend can either be displayed in a `'column'` or `'row'` layout with the `direction` property.

> It can also be moved by a combination of `position: { vertical, horizontal }` and the legend offset values.

Reviewer: ```suggestion
It can also be moved by a combination of `position: { vertical, horizontal }` properties and the legend offset values.
```

> The `position` places the legend tight to the drawing area, and offset values move it relative to this base position.

Author: My idea was to say that the legend is placed at the limit of the chart area, and from this position, spacing can be added with offset.

![image](https://github.com/mui/mui-x/assets/45398769/328b9a69-acb9-4c18-9966-2c99336cb1f6)

For me "tight" means "just next to" but seems it's not 🙈

```suggestion
The `position` places the legend just next to the drawing area, and offset values move it relative to this base position.
```

- `vertical` can be `'top'`, `'middle'`, or `'bottom'`.
- `horizontal` can be `'left'`, `'middle'`, or `'right'`.

> - offsets are set with CSS variables `----Legend-rootOffsetX` and `--Legend-rootOffsetY`.

Reviewer: ```suggestion
- offsets are set with CSS variables `--Legend-rootOffsetX` and `--Legend-rootOffsetY`.
```
